inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -827,3 +830,88 @@ def extract_authors_from_pdf(obj, eng):
                 grobid_authors_count
             )
     return obj
+
+
+def prepare_collaboration_multi_search(collaborations):
+    compiled_regexp = get_collaboration_cleaner_regexp()
+    multi_search = MultiSearch(index="records-experiments", using=current_search_client)
+    backup_multi_search = MultiSearch(index="records-experiments", using=current_search_client)
+    for collaboration in collaborations:
+        full_collaboration_string = collaboration.get('value', '')
+        normalized_collaboration_string = compiled_regexp.sub('', full_collaboration_string).lower().strip()
+        if collaboration.get('record'):
+            # Add dummy search so multisearch will stay in sync with collaborations
+            multi_search = multi_search.add(Search().query().source(False))
+            backup_multi_search = backup_multi_search.add(Search().query().source(False))
+            continue
+
+        multi_search = multi_search.add(build_standard_collaboration_search(full_collaboration_string, normalized_collaboration_string))
+        backup_multi_search = backup_multi_search.add(build_backup_collaboration_search(full_collaboration_string, normalized_collaboration_string))
+    return multi_search, backup_multi_search
+
+
+def build_standard_collaboration_search(full_collaboration_string, normalized_collaboration_string, ):
+    normalized_name_search = Q("term", normalized_name_variants_keyword={"value": full_collaboration_string, "boost": 10})
+    normalized_match = Q("match", normalized_name_variants_text=normalized_collaboration_string)
+
+    query = normalized_name_search | normalized_match
```

Comment:
I don't understand why you'd need to do both searches here.

Reply:
As both of them are giving different results in different situations. First one is more important as it's doing exact match and it's more important (it's boosted) as second one is trying to find anything similar (with words like word "group" removed etc.)